rocAL/source/decoders/image/rocjpeg_decoder.cpp

```diff
@@ -0,0 +1,386 @@
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in
+all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.  IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+THE SOFTWARE.
+*/
+
+
+#include "pipeline/commons.h"
+#include <stdio.h>
+#include <string.h>
+#include "decoders/image/rocjpeg_decoder.h"
+
+#if ENABLE_ROCJPEG
+
+#include "hip/hip_runtime_api.h"
+#include "hip/hip_runtime.h"
+#include "rocal_hip_kernels.h"
+
+#define CHECK_HIP(call) {                                             \
+    hipError_t hip_status = (call);                                   \
+    if (hip_status != hipSuccess) {                                   \
+        std::cerr << "HIP failure: 'status: " << hipGetErrorName(hip_status) << "' at " << __FILE__ << ":" << __LINE__ << std::endl;\
+        exit(1);                                                      \
+    }                                                                 \
+}
+
+#define CHECK_ROCJPEG(call) {                                             \
+    RocJpegStatus rocjpeg_status = (call);                                \
+    if (rocjpeg_status != ROCJPEG_STATUS_SUCCESS) {                       \
+        std::cerr << #call << " returned " << rocJpegGetErrorName(rocjpeg_status) << " at " <<  __FILE__ << ":" << __LINE__ << std::endl;\
+        exit(1);                                                        \
+    }                                                                     \
+}
+
+HWRocJpegDecoder::HWRocJpegDecoder() {
+};
+
+/**
+ * @brief Gets the channel pitch and sizes.
+ *
+ * This function gets the channel pitch and sizes based on the specified output format, chroma subsampling,
+ * output image, and channel sizes.
+ *
+ * @param output_format The output format.
+ * @param subsampling The chroma subsampling.
+ * @param widths The array to store the channel widths.
+ * @param heights The array to store the channel heights.
+ * @param num_channels The number of channels.
+ * @param output_image The output image.
+ * @param channel_sizes The array to store the channel sizes.
+ * @return The channel pitch.
+ */
+int GetChannelPitchAndSizes(RocJpegOutputFormat output_format, RocJpegChromaSubsampling subsampling, uint32_t *widths, uint32_t *heights,
```

Comment:
Please modify the first argument of the `GetChannelPitchAndSizes` function to accept `RocJpegDecodeParams decode_params` instead of `RocJpegOutputFormat output_format`. You can access the `output_format` using `decode_params.output_format` within the function. This change is necessary because `decode_params` also includes the `crop_rectangle`, which is essential for supporting regions of interest (ROI).

Please take a look to the [GetChannelPitchAndSizes](https://github.com/ROCm/rocJPEG/blob/develop/samples/rocjpeg_samples_utils.h#L318) in rocJPEG and use the same changes here.

Reply:
Currently we have not added the support in rocAL to decode images with ROI using rocJpeg decoder, Please let me know if this will be required if the ROI is not updated internally.

Reply:
Done

Reply:
@AryanSalmanpour added the changes to include ROI in GetChannelPitchAndSizes